network/hnswrapper/hnsv2wrapperwithtimeout.go

```diff
@@ -63,13 +80,15 @@ func (h Hnsv2wrapperwithtimeout) CreateEndpoint(endpoint *hcn.HostComputeEndpoin
 	select {
 	case res := <-r:
 		return res.endpoint, res.Err
-	case <-time.After(3 * time.Second):
-		return nil, fmt.Errorf("out of time :(")
+	case <-ctx.Done():
+		return nil, fmt.Errorf("CreateEndpoint %w , timeout value is %s seconds", ErrHNSCallTimeout, hnsCallTimeout.String())
```

Comment:
errors.Wrap/errors.Wrapf (everywhere, instead of fmt.Errorf)

Reply:
Doesn't %w introduced in go 1.13 replace those packages for wrapping the error? I am using %w here to wrap the error with the standard library, I know errors.Wrapf includes the stack trace, is that the extra benefit we are looking to achieve by leveraging this external package github.com/pkg/errors? 

Otherwise these two do the same thing but one uses the standard library.

Reply:
If its just a matter of formatting, wouldn't it be more prudent to create some sort of common method to return errors in the format we want? This way we control the format and we use the standard library and folks don't have to think to much about this. 

I don't have a strong opinion on this and will go with wrapf, but slightly curious.

Reply:
> common method to return errors in the format we want

sure, we could call it something like `pkg/errors.Wrap`? 😄 
`pkg/errors` is commonly used, feature complete, stable, and not going anywhere, I don't see a reason to reimplement it ourselves

Reply:
For me, when I read the go official documentation mentioning %w as the way to wrap errors and in this repo, there is examples of doing that in that way - this leads to confusion for folks that are not aware we started using pkg/errors everywhere. Maybe having this as part of the linter would help? 

In any case, I updated it to use errors.wrapf.